Add "custom classes" section in docs (#11962)
decidim/decidim: docs/modules/develop/pages/classes/commands.adoc

Reviewer: Add credits and link to https://github.com/andypike/rectify in a "More info" section

Author: Fixed in c010e4fed7
![image](https://github.com/decidim/decidim/assets/105683/8ef62934-329f-4580-b51c-4720c81000da)

          event: "decidim.events.my_module.my_resource_created",
          event_class: Decidim::MyModule::MyResourceEvent,
          resource:
        )
    end

    def create_resource
      @resource = Decidim.traceability.create!(
        resource,
        form.current_user,
        **attributes,
        visibility: "public-only"
      )
    end

    # this is mapping of
    # ActiveRecord::attribute => form.attribute
    def attributes
      {
        title: form.title,
        description: form.description,
        resource: form.resource
      }
    end
  end
end
```

Sometimes you may need to extend a `Decidim` supplied command, then you can either override the `attributes` method, either extend it with a `super` call.

```ruby
# frozen_string_literal: true

# app/lib/overrides/commands/create_my_resource.rb
module Decidim
  module Overrides
    module Commands
      module CreateMyResource
        def attributes
          super.merge(
            {
              my_custom_attribute: form.my_custom_attribute
            }
          )
        end
      end
    end
  end
end

Decidim::MyModule::CreateMyResource.prepend(Decidim::Overrides::Commands::CreateMyResource)
```

> == More Info

Author: ```suggestion
== More information
```

= Commands

A command is custom class that manages the logic between your input, usually a xref:develop:classes/forms.adoc[form] object, an object and an user.

Usually, a command is used to create, update or delete a object.

Commands are located in the `app/commands/decidim/<my_plugin>` directory, and named: `<my_command>.rb`.

An example of a command class would:

```ruby
# frozen_string_literal: true

module Decidim
  module MyPlugin
    # A command with the business logic to create a resource.
    class CreateMyResource < Decidim::Command
      # Public: Initializes the command.
      #

>       # form
>       # resource

Author: ```suggestion
```

= Commands

> A command is custom class that manages the logic between your input, usually a xref:develop:classes/forms.adoc[form] object, a object and an actor.

Reviewer: I can't find what's an Actor in the Decidim terminology, maybe it's better to just call them Users? 

```suggestion
A command is custom class that manages the logic between your input, usually a xref:develop:classes/forms.adoc[form] object, an object and an user.
```

Usually, a command is used to create, update or delete a object.

Commands are located in the `app/commands/decidim/<my_plugin>` directory, and named: `<my_command>.rb`.

An example of a command class would:

```ruby
# frozen_string_literal: true

> 
> module Decidim

Reviewer: ```suggestion

# app/commands/decidim/my_plugin/create_my_resource.rb
module Decidim
```

  module MyPlugin

>     # A command with the business logic to invite an user to an organization.

Reviewer: ```suggestion
    # A command with the business logic to create a resource.
```

    class CreateMyResource < Decidim::Command
      # Public: Initializes the command.
      #
      # form

>       # resource

Reviewer: Does it makes sense to have the params as a comment?

Author: not in that form. I need to remove it :)

      def initialize(form, resource)
        @form = form
        @resource = resource
      end

    def call
      return broadcast(:invalid) if form.invalid?

      transation do
         create_resource
      end
      broadcast(:ok)
    end

    private

    attr_reader :form, :resource

    def create_resource
      Decidim.traceability.create!(
        resource,
        form.current_user,
        **attributes,
        visibility: "public-only"
      )
    end

    # this is mapping of
    # ActiveRecord::attribute => form.attribute
    def attributes
      {
        title: form.title,
        description: form.description,
        resource: form.resource
      }
    end
  end
end

> 

Reviewer: ```suggestion
```

```

> 

Reviewer: ```suggestion
```

> Sometimes, you may need to extend a `Decidim` supplied command, then you can either override the `attributes` method, either extend it with a `super` call.

Reviewer: I think all the overrides should be in the same page, probably https://docs.decidim.org/en/v0.27/customize/code, and then reference from the classes documentation to that page. The other way around would be OK to me, as long as it's consistent between the different strategies and classes.  

```suggestion
Sometimes you may need to extend a `Decidim` supplied command, then you can either override the `attributes` method, either extend it with a `super` call.
```

```ruby

> module Decidim

Reviewer: ```suggestion
# frozen_string_literal: true

# app/overrides/commands/create_my_resource.rb
module Decidim
```
